File: frontend/src/api/learningPath.ts
import { http } from './http'

const LEARNING_PATH_TIMEOUT_MS = 120_000

export interface LearningPathTask {
  day: number
  knowledge_point: string
  task_type: string
  questions: string[]
  description: string
}

export interface LearningPathMilestone {
  day: number
  description: string
  checkpoints: string[]
}

export interface LearningPath {
  student_id: string
  priority_order: string[]
  daily_tasks: LearningPathTask[]
  milestones: LearningPathMilestone[]
  estimated_days: number
  generated_at: string
  source: string
}

export function fetchLearningPath(studentId: string) {
  return http.get<LearningPath>(`/profile/${studentId}/learning-path`, {
    timeout: LEARNING_PATH_TIMEOUT_MS,
  })
}

File: backend/app/schemas/learning_path.py
from pydantic import BaseModel


class LearningPathTask(BaseModel):
    day: int
    knowledge_point: str
    task_type: str = "练习"
    questions: list[str] = []
    description: str


class LearningPathMilestone(BaseModel):
    day: int
    description: str
    checkpoints: list[str] = []


class LearningPath(BaseModel):
    student_id: str
    priority_order: list[str]
    daily_tasks: list[LearningPathTask]
    milestones: list[LearningPathMilestone]
    estimated_days: int = 7
    generated_at: str = ""
    source: str = "rule"
